File: backend/src/main/java/com/backend/model/Productos.java
package com.backend.model;


import jakarta.persistence.*;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;

import lombok.*;


@Entity
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@ToString
@Table(name = "productoss")
@Builder
public class Productos {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String name;

    private String description;

    @Column(nullable = false, precision = 10)
    private Double price;

    private String photoUrl;

    private Boolean descuento;

    @ManyToOne
    @JoinColumn(name = "categoria_id")
    private Category category;

    @ManyToOne
    @JoinColumn(name = "store_id")
    private Store store;
}

File: frond/src/app/model/productos.ts

// Definimos la interfaz
export interface Producto{
   
    id: number;

    name: string;

    description: string;

    price: number;

    photoUrl: string;

    descuento: boolean;

    category: {
        id: number;
        name: string;
    };

    store: {
        id: number;
        name: string;
        location: string;
    };
}



